docs/decisions/0005-credly-xblock.rst

```diff
@@ -0,0 +1,286 @@
+   )
+   js_urls = [
+      'public/js/vendor/virtual-dom-1.3.0.min.js',
+      'public/js/drag_and_drop.js',
+   ]
+
+   statici18n_js_url = self._get_statici18n_js_url()
+   if statici18n_js_url:
+      js_urls.append(statici18n_js_url)
+
+   for css_url in css_urls:
+      fragment.add_css_url(self.runtime.local_resource_url(self, css_url))
+   for js_url in js_urls:
+      fragment.add_javascript_url(self.runtime.local_resource_url(self, js_url))
+
+   self.include_theme_files(fragment)
+
+   fragment.initialize_js('DragAndDropBlock', self.student_view_data())
+
+Library source code: https://github.com/openedx/web-fragments
+XBlock source code: https://github.com/openedx/xblock-drag-and-drop-v2/blob/master/drag_and_drop_v2/drag_and_drop_v2.py
+
+Badgr XBlock
+============
+
+The badgr-xblock was developed to work in conjunction with the
+open-source Badgr Server application or the hosted version at Badgr.io.
+The badgr-xblock communicates with the Badgr API, and awards badges
+based on a passing grade for a specified subsection in a course.
+
+To do this, the badgr-xblock uses various XBlock services, such as
+settings, badging, and user services, it also executes various API calls
+to edx-platform grades API v0 (which no longer exists on versions newer
+than Ginkgo) and the Badgr API. Also on the XBlock, a few fields are
+defined to let the staff set up which badge to award, and to which
+subsection will determine if a score is enough to assert a badge,
+here is the list of fields:
+
+-  issuer_slug: Issuer name.
+-  badge_slug: Badge name.
+-  badge_name: Badge name that appears in the Accomplishments tab.
+-  image_url: The URL for the badge image on the Badgr server.
+-  criteria: Text to describe how one earns this badge.
+-  description: Text description of this badge.
+-  section_title: This should be the display name of the sub-section you
+   want to check the score from.
+-  pass_mark: Minimum grade required to award this badge.
+-  received_award: Boolean to store if the user received a badge for
+   this sub-section.
+-  check_earned: True if they are eligible for a badge.
+-  assertion_url: The URL of the asserted badge in case it was already
+   awarded.
+-  award_message: Message the user will see upon receiving a badge.
+-  motivation_message: Message the user will see if they do not qualify
+   for a badge.
+
+Once setup, this XBlock will display a button with the text “Click here
+to view your results.”, once the learner clicks on it, the JavaScript
+function getGrades will be executed to request the current grading of
+the setup sub-section, if the minimum pass mark is achieved, it will
+execute the method new_award_badge using the XBlock handlerUrl runtime
+to award a new badge using the current setup badging service set up on
+the edx-platform, and reload the XBlock HTML to display the newly
+created badge for this user, otherwise, it will reload the HTML to an
+error message.
+
+XBlock Proposal
```

Comment:
This proposal clearly defines the events for which the xblock would award badges. Could you please provide a brief integration or logic workflow with Credly?. Credly and Badgr are not exactly the same, therefore as a result of this fact the workflow might be slightly different. For example: Badgr allows us to create BadgeClass while Credly apparently does not, which means that for Credly the Badges would need to be created manually on Credly side and once staff/content creators are setting up the Xblock, this Xblock could get all available Badges to link the course with an specifc one. @kuipumu

Reply:
This xblock will rely on the Credly implementation we would create for the platform, so this block does not need to interact directly with Credly API. @Jacatove So it's worth mentioning that this block will depend on the Credly backend implementation @kuipumu

Reply:
@Jacatove I haven't checked if templates can be created using the Credly API. We should let the staff decide which template to use, so it would be logical to let staff set on the XBlock settings the template to use. @Squirrel18 I will add a reference to PADV-234 so we can have a reference on how the API logic workflow works with Credly.

Reply:
@Squirrel18 I got it, I was missing that link.

Reply:
@kuipumu There is no such API and the way I see things from current approach of this Xblock relying on a Credly Badging system, therefore my concern should not be a issue here as result of that when using this Xbloxk, then the Badging system should guarantee everything is working just fine. Therefore my concern will need to be addressed in the other discovery.

Reply:
@Jacatove We should not have a problem with the XBlock approach if we follow the same approach of badgr-xblock and set the badge_slug (template slug) to be used on badges (assertions) in the XBlock fields.